homeassistant/components/lupusec/entity.py

```diff
@@ -0,0 +1,48 @@
+"""Provides the Lupusec entity for Home Assistant."""
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+
+class LupusecDevice(Entity):
+    """Representation of a Lupusec device."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, data, device, config_entry) -> None:
+        """Initialize a sensor for Lupusec device."""
+        self._data = data
+        self._device = device
+        self._entry_id = config_entry.entry_id
+        self._attr_unique_id = self.get_unique_id(
+            config_entry.entry_id, device.device_id
+        )
+
+    def update(self):
+        """Update automation state."""
+        self._device.refresh()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._device.name
```

Comment:
Right, remove this clause and add `_attr_name = None` to all entities. Having the combi of `_attr_has_entity_name = True` and `_attr_name = None` will make the entity follow the device name.

Reply:
Done.